### PR TITLE
330 add sortedby and order to all project endpoints

### PR DESCRIPTION
**Pull request by ywkim312**
New sortedBy and order parameters added to the following project endpoints 

- GET /project/api/projects
- GET project/api/projects/{project-id}/hazards
- GET project/api/projects/{project-id}/visualizations
- GET project/api/projects/{project-id}/datasets
- GET project/api/projects/{project-id}/workflows
- GET project/api/projects/{project-id}/mappings

Default to sort by “Date”
Default to sort by descending order. Latest items show up first

How to test
- http://localhost:8080/project/api/projects?sortBy=date&order=desc (the result should be identical without any sortBy option since it is default. But you can change it to asec to test it)
- http://localhost:8080/project/api/projects/673b789f74c21f67fa69003b/datasets?sortBy=title&order=desc
- http://localhost:8080/project/api/projects/673b789f74c21f67fa69003b/hazards?sortBy=creator&order=desc

There are other options and endpoints so please test various options.

### CODE REVIEW

**Review by Rashmil-1999 — APPROVED** [review]

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java [hunk 1]
```diff
@@ -109,6 +109,16 @@ public void removeSourceDatasetId(String id) {
...
     }
 
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
```
Comment (navarroc) [inline server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java:119]: Can you remove the extra line break here?

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/utils/CommonUtil.java [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+        }
+
+        if ("desc".equalsIgnoreCase(order)) {
+            comparator = comparator.reversed();
+        }
+
+        return comparator;
+    }
+
+
```
Comment (navarroc) [inline server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/utils/CommonUtil.java:173]: Can you remove the extra line break here?

**Comment by navarroc** [comment]
Change looks good, just minor style nitpicks and then I think it's ready to merge

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/utils/CommonUtil.java (hunk 2 above)
Comment (ywkim312) [reply server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/utils/CommonUtil.java:173]: removed

File: server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java (hunk 1 above)
Comment (ywkim312) [reply server/project-service/src/main/java/edu/illinois/ncsa/incore/service/project/models/VisualizationResource.java:119]: removed

**Review by navarroc — APPROVED** [review]

**Pull request merged**